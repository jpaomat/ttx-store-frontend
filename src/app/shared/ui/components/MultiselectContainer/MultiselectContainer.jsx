import PropTypes from 'prop-types';
import './MultiselectContainer.scss';
import { useEffect } from 'react';
import Multiselect from 'multiselect-react-dropdown';

export const MultiselectContainer = ({selector, label, options, selectedValues, displayValue, onSelect, setSelectorDefaultValues}) => {
	useEffect(() => {
		setSelectorDefaultValues((currentValue) => ({
			...currentValue,
			...{ [`${selectedValues.codeName}`]: selectedValues.code },
		}));
	}, []);

	return (
		<div className='select-actions' key={selector}>
			<label className='select-actions__label'>{label}</label>
			<Multiselect
				className='select-actions__select'
				options={options}
				singleSelect= {true}
				selectedValues={[selectedValues]}
				displayValue={displayValue}
                onSelect={onSelect}
			/>
		</div>
	);
};

MultiselectContainer.defaultProps = {
    selector: '',
	label: '',
	options: [{}],
	selectedValues: {
		code: 0,
		codeName: '',
		name: '',
	}
};

MultiselectContainer.propTypes = {
	selector: PropTypes.string,
	label: PropTypes.string.isRequired,
	options: PropTypes.arrayOf(PropTypes.shape({})),
	displayValue: PropTypes.string,
	selectedValues: PropTypes.shape({
		code: PropTypes.number,
		codeName: PropTypes.string,
		name: PropTypes.string,
	}),
    onSelect: PropTypes.func,
	setSelectorDefaultValues: PropTypes.func,
};
