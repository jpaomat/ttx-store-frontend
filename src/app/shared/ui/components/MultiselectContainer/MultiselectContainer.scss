@import '../../../../../assets/scss/';

.select-actions {
	margin: 4px;
	&__label {
		margin: 6px;
	}
	&__select {
		border-radius: 8px;
		border: 1px solid $purple-500;
	}

	.singleSelect {
		padding: 4px 10px;
	}

	.chip {
		margin: 0;
		padding: 0;
	}

	.multiSelectContainer {
		ul {
			max-height: 120px;
		}
		.optionContainer {
			li:hover,
			.highlight {
				background: $purple-100;
				color: $black;
			}
			li {
				padding: 8px;
				border-bottom: 1px solid $gray-150;
			}
		}
	}
}
