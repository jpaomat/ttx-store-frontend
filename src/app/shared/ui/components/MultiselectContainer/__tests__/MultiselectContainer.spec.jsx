import { render, screen } from '@testing-library/react';
import { MultiselectContainer } from '../MultiselectContainer';
import { defaultMock } from './mocks';

describe('<MultiselectContainer /> component', () => {
	test('should render component', () => {
		const { container } = render(<MultiselectContainer {...defaultMock}/>);
		expect(container).toMatchSnapshot();
	});

    it('should render component and validate label', () => {
		render(<MultiselectContainer {...defaultMock}/>);
		expect(screen.getByText('Multiselect test')).toBeTruthy();
	});

    it('should render component and show the number of options that are sent in the props', () => {
		const {container} = render(<MultiselectContainer {...defaultMock}/>);
		expect(container.querySelectorAll('.option').length).toEqual(defaultMock.options.length);
	});
});
