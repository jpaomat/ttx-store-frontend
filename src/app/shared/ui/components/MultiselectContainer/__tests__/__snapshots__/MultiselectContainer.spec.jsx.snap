// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiselectContainer /> component should render component 1`] = `
<div>
  <div
    class="select-actions"
  >
    <label
      class="select-actions__label"
    >
      Multiselect test
    </label>
    <div>
      <div
        class="multiselect-container multiSelectContainer  select-actions__select"
        id="multiselectContainerReact"
      >
        <div
          class="search-wrapper searchWrapper singleSelect"
        >
          <span
            class="chip  singleChip false"
          >
            <img
              class="icon_cancel closeIcon"
              src="data:image/svg+xml,%3Csvg%20viewBox%3D%220%200%2096%2096%22%20%20%20%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%20%20%20%20%3Cstyle%20type%3D%22text%2Fcss%22%3E%20%20%20%20%20%20%20%20.st0%7B%20%20%20%20%20%20%20%20%20%20%20%20fill%3A%23fff%3B%20%20%20%20%20%20%20%20%7D%20%3C%2Fstyle%3E%20%20%20%20%3Cg%3E%20%20%20%20%20%20%20%20%3Cpath%20class%3D%22st0%22%20d%3D%22M48%2C0A48%2C48%2C0%2C1%2C0%2C96%2C48%2C48.0512%2C48.0512%2C0%2C0%2C0%2C48%2C0Zm0%2C84A36%2C36%2C0%2C1%2C1%2C84%2C48%2C36.0393%2C36.0393%2C0%2C0%2C1%2C48%2C84Z%22%2F%3E%20%20%20%20%20%20%20%20%3Cpath%20class%3D%22st0%22%20d%3D%22M64.2422%2C31.7578a5.9979%2C5.9979%2C0%2C0%2C0-8.4844%2C0L48%2C39.5156l-7.7578-7.7578a5.9994%2C5.9994%2C0%2C0%2C0-8.4844%2C8.4844L39.5156%2C48l-7.7578%2C7.7578a5.9994%2C5.9994%2C0%2C1%2C0%2C8.4844%2C8.4844L48%2C56.4844l7.7578%2C7.7578a5.9994%2C5.9994%2C0%2C0%2C0%2C8.4844-8.4844L56.4844%2C48l7.7578-7.7578A5.9979%2C5.9979%2C0%2C0%2C0%2C64.2422%2C31.7578Z%22%2F%3E%20%20%20%20%3C%2Fg%3E%3C%2Fsvg%3E"
            />
          </span>
          <input
            autocomplete="off"
            class="searchBox display-none"
            disabled=""
            id="search_input"
            name="search_name_input"
            placeholder=""
            type="text"
            value=""
          />
          <img
            class="icon_cancel icon_down_dir"
            src="data:image/svg+xml,%3Csvg%20height%3D%2232%22%20viewBox%3D%220%200%2032%2032%22%20width%3D%2232%22%20%20%20%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%20%20%20%20%3Cg%20id%3D%22background%22%3E%20%20%20%20%20%20%20%20%3Crect%20fill%3D%22none%22%20height%3D%2232%22%20width%3D%2232%22%2F%3E%20%20%20%20%3C%2Fg%3E%20%20%20%20%3Cg%20id%3D%22arrow_x5F_down%22%3E%20%20%20%20%20%20%20%20%3Cpolygon%20points%3D%222.002%2C10%2016.001%2C24%2030.002%2C10%20%20%22%2F%3E%20%20%20%20%3C%2Fg%3E%3C%2Fsvg%3E"
          />
        </div>
        <div
          class="optionListContainer displayNone"
        >
          <ul
            class="optionContainer"
          >
            <li
              class="option  highlightOption highlight  "
            >
              Test 1
            </li>
            <li
              class="option    "
            >
              Test 2
            </li>
            <li
              class="option    "
            >
              Test 3
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
